docs/dev/contributing/backend/service-objects.md

```diff
@@ -118,71 +118,72 @@ func (c formCreator) CreateForm(template services.FormTemplate) (afero.File, err
+object that has dependencies:
 
-// FormCreator is the service object interface for CreateForm
-type FormCreator interface {
-  CreateForm(template FormTemplate) (afero.File, error)
+```golang
+type shipmentRejecter struct {
+  router services.ShipmentRouter
 }
-
 ```
 
-### Naming and Defining Service Object Execution Method
+2. Add an interface for the service, that captures the behavior of the service object.
+This will go in the appropriate file under `pkg/services`. For example, all
+shipment-related service objects live in `pkg/services/mto_shipment.go`. Here's
+an example of an interface you might add there:
 
-The service object execution method is responsible for kicking off the service object call. Ideally, the service object
-should expose only one public function, with helper private functions, as needed and when it makes sense. Oftentimes,
-smaller private functions are good to unit test smaller units of functionality. The service object execution method should be the same as the file name
-and struct. The service object execution method should be a method of the service object struct,
-a struct of parameters that the service object requires, and returning values, as appropriate.
+```golang
+//ShipmentRejecter is the service object interface for rejecting a shipment
+//go:generate mockery --name ShipmentRejecter --disable-version-string
+type ShipmentRejecter interface {
+  RejectShipment(appCtx appcontext.AppContext, shipmentID uuid.UUID, eTag string, reason *string) (*models.MTOShipment, error)
+}
+```
 
-```go
-// form_creator.go
-package paperwork
+### Naming and Defining Public Service Object Functions
 
-import (
-    "github.com/spf13/afero"
-    "github.com/transcom/mymove/pkg/services"
-)
+Ideally, the service object should expose only one public function, which should
+be as short as possible, with its various actions broken down into smaller
+private functions. The service object execution method should be a method of the
+service object struct, and **its first argument should always be of the type
+`appcontext.AppContext`**. For example:
 
-type createForm struct {
-  fileStorer FileStorer
-  formFiller FormFiller
-}
+```golang
+// pkg/services/mto_service_item/mto_service_item_updater.go
+func (p *mtoServiceItemUpdater) ApproveOrRejectServiceItem(appCtx appcontext.AppContext, mtoServiceItemID uuid.UUID, status models.MTOServiceItemStatus, rejectionReason *string, eTag string) (*models.MTOServiceItem, error) {
+  mtoServiceItem, err := p.findServiceItem(appCtx, mtoServiceItemID)
+  if err != nil {
+    return &models.MTOServiceItem{}, err
+  }
 
-func (c formCreator) CreateForm(template services.FormTemplate) (afero.File, error) {
-  ...
+  return p.approveOrRejectServiceItem(appCtx, *mtoServiceItem, status, rejectionReason, eTag, checkMoveStatus(), checkETag())
 }
 ```
+`appCtx` is how we pass around the DB connection, logger, and session from a
+handler to the service object.
 
```

Comment:
I know your PR is focused on appCtx, so I just want to add a suggestion here. I think it will be helpful to add a comment about what the service object should return. The rule of thumb Roci was using was that a service object (SO) returns reusable services. A lot of times we saw SOs that returned an entire shipment if that SO was related to a shipment (e.g. SIT extension, agents, etc), and we want to steer other engineers away from returning a shipment. If you have time to add some information about this that would be a nice to have, but not a requirement for this PR.

Reply:
That's interesting. I probably won't have time in this PR. It might need more conversation with other teams. I know on TTV that we returned model types from service object functions, and then we defined the payload in the `model_to_payload` file to match what the front end needed.